### ISSUE TITLE
Password Protection

### ISSUE THREAD

**Issue by ItsRelative** [giowck/symphytum#99]
I'd like to restrict access to my Symphytum data by using a password. Any way to do this?

**Comment by joshirio** [comment]
No, that is not possible. To be honest, it's very hard to implement this securely because Symphytum would have to decrypt the database before using it, which involves writing the sqlite database file to your disk anyway so that it is not really secure. Of course, there could be a way to load the database and decrypt in only in RAM but that is a huge amount of work and would increase the resource usage greatly. Symphytum uses SQLite as underlying database technology. SQLite has a premium feature which includes in memory encryption/decryption but that is not part of the open source distribution and is a commercial feature.

A much better solution which is also much more secure is to enable full disk encryption for your hard drive in combination with screen locking. Windows has BitLocker or VeraCrypt, macOS has FileVault and Linux LUKS.

An alternative would be just to ask for a password when opening the database) without involving any encryption) but any tech savvy user would come around this measure very quickly.

So I think this feature is out of scope, especially when much more secure solutions already exist.

**Comment by ItsRelative** [comment]
Thanks for the clarification. 
How would I set up requiring a password when opening the database?
